### ISSUE TITLE
Deleting the title of a photo leaves a ghost behind

### ISSUE THREAD

**Issue by dssinger** [RhetTbull/osxphotos#512]
I ran across this while testing the fix for #506.  If I remove the title from a photo, the '{title?}' test still is true but '{title,}' expands to an empty string.

I'm attaching a one-photo library in three different states:

[before.zip](https://github.com/RhetTbull/osxphotos/files/7071718/before.zip) - this is an empty library with one photo imported; it has neither a title nor a description.  Exporting the library with a template of `"{title,}{title?{descr? - ,},}{descr,}"` results in no title and no description, as expected.

````
❯ ./titletest
2021-08-28 17:36:16,111 - WARNING - photosdb.py - 100 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4]: you have Darwin, OS version: 11.5
Exporting 1 photo to /Users/david/Desktop/Test...
  [####################################]  100%
Processed: 1 photo, exported: 1, missing: 0, error: 0, touched date: 1
Elapsed time: 0.728 seconds
pizza.jpg
[]              title                           : -
[]              description                     : -
````

[during.zip](https://github.com/RhetTbull/osxphotos/files/7071721/during.zip) - this is the same library after giving the photo a title and a description (using the info pane in Photos).  Exporting the library with the same template gives the expected title and description.

````
2021-08-28 17:41:35,855 - WARNING - photosdb.py - 100 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4]: you have Darwin, OS version: 11.5
Exporting 1 photo to /Users/david/Desktop/Test...
  [####################################]  100%
Processed: 1 photo, exported: 1, missing: 0, error: 0, touched date: 1
Elapsed time: 0.981 seconds
pizza.jpg
[XMP]           Title                           : Pizza
[XMP]           Description                     : Pizza - From Otto Portland
/Users/david/Desktop/Test/pizza.jpg:
findercomment               com.apple.metadata:kMDItemFinderComment            = Pizza - From Otto Portland
````

[after.zip](https://github.com/RhetTbull/osxphotos/files/7071724/after.zip) - this is the same library after backspacing over the title so it's empty.  The template expands to ' - From Otto Portland'.

````
2021-08-28 17:42:48,515 - WARNING - photosdb.py - 100 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4]: you have Darwin, OS version: 11.5
Exporting 1 photo to /Users/david/Desktop/Test...
  [####################################]  100%
Processed: 1 photo, exported: 1, missing: 0, error: 0, touched date: 1
Elapsed time: 1.122 seconds
pizza.jpg
[]              title                           : -
[XMP]           Description                     : - From Otto Portland
/Users/david/Desktop/Test/pizza.jpg:
findercomment               com.apple.metadata:kMDItemFinderComment            = - From Otto Portland
````
Here's the script I used: 
[titletest.zip](https://github.com/RhetTbull/osxphotos/files/7071728/titletest.zip)

Thanks.

**Comment by RhetTbull** [comment]
Interesting -- I checked the Photos database and for a photo that's never had a title set, Photos sets the title column (`ZADDITIONALASSETATTRIBUTES.ZTITLE`) to `NULL` which maps to `None` in python.  BUT -- if you delete the title, as in your example, Photos sets the field to an empty string.  osxphotos must be checking for `value is not None` which is false in the case of an empty string.

**Comment by RhetTbull** [comment]
And, boom! I think it's fixed:

Before:
```
Exporting 1 photo to /Users/rhet/Desktop/Test...
  [####################################]  100%
Processed: 1 photo, exported: 1, missing: 0, error: 0, touched date: 1
Elapsed time: 1.508 seconds
pizza.jpg
[XMP]           Title                           : Pizza
[XMP]           Description                     : Pizza - From Otto Portland
/Users/rhet/Desktop/Test/pizza.jpg:
findercomment               com.apple.metadata:kMDItemFinderComment            = Pizza - From Otto Portland
```

After:
```
[I] ➜ sh ~/Downloads/titletest
Exporting 1 photo to /Users/rhet/Desktop/Test...
  [####################################]  100%
Processed: 1 photo, exported: 1, missing: 0, error: 0, touched date: 1
Elapsed time: 1.253 seconds
pizza.jpg
[]              title                           : -
[XMP]           Description                     : From Otto Portland
/Users/rhet/Desktop/Test/pizza.jpg:
findercomment               com.apple.metadata:kMDItemFinderComment            = From Otto Portland
```

Need to finish testing.  Interestingly Photos does not do this for the description, only the title.  I'll also update PhotoInfo.title to return None if there's an empty string as this is what the other fields do.

**Comment by RhetTbull** [comment]
I think this is fixed in [v0.42.79](https://github.com/RhetTbull/osxphotos/releases/tag/v0.42.79)

**Comment by dssinger** [comment]
It's fixed, thanks!
